cdm/core/src/main/java/thredds/inventory/CollectionSpecParser.java

```diff
@@ -132,6 +86,67 @@ public CollectionSpecParser(String rootDir, String regExp, Formatter errlog) {
     this.filterOnName = false;
   }
 
+  private static String getRootDir(String collectionSpec) {
+    if (collectionSpec.contains("/**/")) {
+      return collectionSpec.split(Pattern.quote("/**/"))[0];
+    }
+
+    if (collectionSpec.contains("/")) {
+      return collectionSpec.substring(0, collectionSpec.lastIndexOf('/'));
+    }
+
+    return System.getProperty("user.dir"); // working directory
+  }
+
+  private static String getFilterAndDateMarkString(String collectionSpec) {
+    final int posFilter =
+        collectionSpec.contains("/**/") ? collectionSpec.indexOf("/**/") + 3 : collectionSpec.lastIndexOf('/');
+
+    if (posFilter >= collectionSpec.length() - 2) {
+      return null;
+    }
+
+    return collectionSpec.substring(posFilter + 1); // remove topDir
+  }
+
+  private static Pattern getRegEx(String filterAndDateMark) {
+    if (filterAndDateMark == null) {
+      return null;
+    }
+
+    int numberOfHashes = filterAndDateMark.length() - filterAndDateMark.replace("#", "").length();
+
+    if (numberOfHashes == 0) {
+      return Pattern.compile(filterAndDateMark);
+    } else if (numberOfHashes == 1) {
+      return Pattern.compile(filterAndDateMark.substring(0, filterAndDateMark.indexOf('#')) + "*");
+    } else {
+      StringBuilder sb = new StringBuilder(StringUtil2.remove(filterAndDateMark, '#')); // remove hashes, replace with .
+
+      for (int i = filterAndDateMark.indexOf('#'); i < filterAndDateMark.lastIndexOf('#') - 1; i++) {
+        sb.setCharAt(i, '.');
+      }
+
```

Comment:
can this just be `filterAndDateMark.replace("#", ".")`?

Reply:
I think that comment on line 124 is a bit misleading. We want to replace the whole date part with  "."-- So if you have `directory/foo#yyyyMMdd_HHmm#.*grib1` it assumes the part between the first # and the last # is the date and replaces that with periods in the filter to make `regEx=foo..............*grib1`

didn't see a one liner but I tried to make it more readable in 9fa6453c7f